core/src/mindustry/type/UnitType.java

```diff
@@ -171,6 +171,27 @@ public void display(Unit unit, Table table){
                 bars.add(new Bar(ammoType.icon + " " + Core.bundle.get("stat.ammo"), ammoType.barColor, () -> unit.ammo / ammoCapacity));
                 bars.row();
             }
+
+            if(unit instanceof Payloadc payload){
+                bars.add(new Bar("stat.payloadcapacity", Pal.items, () -> payload.payloadUsed() / unit.type().payloadCapacity));
+                bars.row();
+
+                bars.table(req -> {
+                    req.top().left();
+
+                    float pad = 1;
+                    float items = payload.payloads().size;
```

Comment:
The table should be rebuilt whenever the payload count changes. This isn't straightforward to implement with Java's `effectively final` system, but what you need to do is store the count in a 1-len array and run a local runnable to rebuild when it changes.